ArduPlane/tailsitter.cpp

```diff
@@ -62,6 +62,61 @@ void QuadPlane::tailsitter_output(void)
     plane.pitchController.reset_I();
     plane.rollController.reset_I();
 
+    static float orig_kP = -1.0f;
+
+    if (tailsitter.tcomp_gain > 0) {
+        if (orig_kP < 0.0f) {
+            orig_kP = attitude_control->get_rate_pitch_pid().kP();
+        }
+        // boost throttle when pitch or yaw error is large
+        float pitch_error_cd = (plane.nav_pitch_cd - ahrs_view->pitch_sensor) * 0.5;
+        float pitch_error_norm = constrain_float(fabsf(pitch_error_cd), 0, 4500) / 4500.0;
+
+        float elevon_L = SRV_Channels::get_output_norm(SRV_Channel::k_elevon_left);
+        float elevon_R = SRV_Channels::get_output_norm(SRV_Channel::k_elevon_right);
+
```

Comment:
getting the elevon outputs won't work on all tailsitters - for example it doesn't work properly on a 3D plane. Instead you ask for the k_aileron and k_elevator and check for saturation on those then it will work for all aircraft, as elevons are derived from those.